tests/unit/components/Player/ItemModal.spec.js

```diff
@@ -4,249 +4,389 @@ import ItemModal from "@/components/Player/ItemModal";
-      { answer: "abc" },
-    ];
-    const proceedQuestionMock = jest.spyOn(
-      ItemModal.methods,
-      "proceedQuestion"
-    );
-    const wrapper = mount(ItemModal, {
-      props: {
-        itemList: clonedeep(global.dummyItems),
-        itemDetailList: clonedeep(global.dummyItemDetails),
-        responseList: responseList,
-        selectedItemIndex: 1,
-      },
-    });
-    wrapper
-      .find('[data-test="footer"]')
-      .find('[data-test="proceedButton"]')
-      .trigger("click");
-
-    expect(proceedQuestionMock).toHaveBeenCalled();
-    expect(wrapper.emitted()).toHaveProperty("proceed-question");
+  describe("checkbox questions", () => {
+    const itemIndex = 1;
+
+    let checkboxItemDetails = clonedeep(global.dummyItemDetails);
+    checkboxItemDetails[itemIndex]["type"] = "checkbox";
+    checkboxItemDetails[itemIndex]["correct_answer"] = [1];
+
+    const mountWrapper = (responseList = null) => {
+      if (responseList == null) {
+        responseList = [];
+        global.dummyItems.forEach(() => {
+          responseList.push({
+            answer: null,
+          });
+        });
+      }
+
+      wrapper = mount(ItemModal, {
+        props: {
+          itemList: clonedeep(global.dummyItems),
+          itemDetailList: checkboxItemDetails,
+          responseList: responseList,
+          selectedItemIndex: itemIndex,
+        },
+      });
+    };
+
+    beforeEach(() => {
+      mountWrapper();
+    });
+
+    afterEach(() => {
+      wrapper.unmount();
+    });
+
+    it("extracts item details correctly", async () => {
+      expect(wrapper.vm.isQuestionTypeSubjective).toBe(false);
+      expect(wrapper.vm.isQuestionTypeMCQ).toBe(false);
+      expect(wrapper.vm.isQuestionTypeCheckbox).toBe(true);
+    });
+
+    it("selecting option makes answer valid", async () => {
+      // initially answer should be invalid
+      expect(wrapper.vm.isAnswerValid).toBeFalsy();
+
+      // select an option
+      await wrapper.find('[data-test="option-0"]').trigger("click");
+
+      // the answer should now be valid
+      expect(wrapper.vm.isAnswerValid).toBeTruthy();
+    });
+
+    it("unselecting all options makes the answer invalid", async () => {
+      // select an option
+      await wrapper.find('[data-test="option-0"]').trigger("click");
+      // select another option
+      await wrapper.find('[data-test="option-1"]').trigger("click");
+
+      // unselect an option
+      await wrapper.find('[data-test="option-0"]').trigger("click");
+      // the answer should still be valid
+      expect(wrapper.vm.isAnswerValid).toBeTruthy();
+
+      // unselect the other option
+      await wrapper.find('[data-test="option-1"]').trigger("click");
+      // the answer should now be invalid
+      expect(wrapper.vm.isAnswerValid).toBeFalsy();
+    });
+
+    it("submits question", async () => {
+      const submitQuestion = jest.spyOn(ItemModal.methods, "submitQuestion");
+      mountWrapper();
```

Comment:
Why are you mounting here again when there's mounting happening in the beforeEach of this test suite?